crates/vm/levm/src/memory.rs

```diff
@@ -25,61 +25,54 @@ impl Memory {
     }
 
     fn resize(&mut self, offset: usize) {
-        if offset.next_multiple_of(32) > self.data.len() {
-            self.data.resize(offset.next_multiple_of(32), 0);
+        if offset.next_multiple_of(WORD_SIZE) > self.data.len() {
+            self.data.resize(offset.next_multiple_of(WORD_SIZE), 0);
         }
     }
 
     pub fn load(&mut self, offset: usize) -> Result<U256, VMError> {
-        self.resize(
-            offset
-                .checked_add(32)
-                .ok_or(VMError::MemoryLoadOutOfBounds)?,
-        );
+        self.resize(offset.checked_add(WORD_SIZE).ok_or(VMError::Internal(
+            InternalError::ArithmeticOperationOverflow, // MemoryLoadOutOfBounds?
+        ))?);
         let value_bytes = self
             .data
             .get(
                 offset
-                    ..offset
-                        .checked_add(32)
-                        .ok_or(VMError::MemoryLoadOutOfBounds)?,
+                    ..offset.checked_add(WORD_SIZE).ok_or(VMError::Internal(
+                        InternalError::ArithmeticOperationOverflow, // MemoryLoadOutOfBounds?
+                    ))?,
             )
             .ok_or(VMError::MemoryLoadOutOfBounds)?;
 
         Ok(U256::from_big_endian(value_bytes))
     }
 
     pub fn load_range(&mut self, offset: usize, size: usize) -> Result<Vec<u8>, VMError> {
-        self.resize(
-            offset
-                .checked_add(size)
-                .ok_or(VMError::MemoryLoadOutOfBounds)?,
-        );
+        let size_to_load = offset.checked_add(size).ok_or(VMError::Internal(
+            InternalError::ArithmeticOperationOverflow,
+        ))?;
+        self.resize(size_to_load);
         self.data
-            .get(
-                offset
-                    ..offset
-                        .checked_add(size)
-                        .ok_or(VMError::MemoryLoadOutOfBounds)?,
-            )
+            .get(offset..size_to_load)
             .ok_or(VMError::MemoryLoadOutOfBounds)
             .map(|slice| slice.to_vec())
     }
 
     pub fn store_bytes(&mut self, offset: usize, value: &[u8]) -> Result<(), VMError> {
         let len = value.len();
-        self.resize(
-            offset
-                .checked_add(len)
-                .ok_or(VMError::MemoryLoadOutOfBounds)?,
-        );
-        self.data.splice(
-            offset
-                ..offset
-                    .checked_add(len)
-                    .ok_or(VMError::MemoryLoadOutOfBounds)?,
-            value.iter().copied(),
-        );
+        let size_to_store = offset.checked_add(len).ok_or(VMError::Internal(
+            InternalError::ArithmeticOperationOverflow,
+        ))?;
+        /*
+               let data_len = self.data.len();
+               if data_len < offset || data_len < size_to_store {
+                   return Err(VMError::MemoryStoreOutOfBounds);
+               }
+        */
```

Comment:
What is this? Should it be removed?

Reply:
In the merges I saw that these checks were not in the 'main' version, so I commented them, they are probably going to be eliminated.